lighthouse-core/audits/dobetterweb/uses-http2.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audit a page to ensure that resource loaded over its own
+ * origin are over the http/2 protocol.
+ * @author Eric Bidelman
+ */
+
+'use strict';
+
+const url = require('url');
+const Audit = require('../audit');
+const Formatter = require('../../formatters/formatter');
+
+class UsesHTTP2Audit extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'uses-http2',
+      description: 'Site uses HTTP/2 for its own resources',
+      helpText: 'h2 offers many benefits over its http/1.1 predecessor: binary, multiplexing, server push, etc. See <a href="https://http2.github.io/faq/" target="_blank">this FAQ</a> for more information.',
+      requiredArtifacts: ['URL', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    if (typeof artifacts.networkRecords === 'undefined' ||
+        artifacts.URL === 'undefined') {
+      return UsesHTTP2Audit.generateAuditResult({
+        rawValue: -1,
+        debugString: 'Network or URL gatherer did not run'
+      });
+    }
+
+    const networkRecords = artifacts.networkRecords[Audit.DEFAULT_PASS];
+    const finalHost = url.parse(artifacts.URL.finalUrl).host;
+
+    // Filter requests that are on the same origin as the page and not over h2.
+    const resources = networkRecords.filter(record => {
+      const requestHost = url.parse(record._url).host;
+      const sameOrigin = requestHost === finalHost;
+      const notH2 = /HTTP\/[01][\.\d]?/i.test(record.protocol);
```

Comment:
I figured ignoring everything but http/[01].x would be ok. For all intents and purposes, quic/spdy is like h2 (more modern protocol and 👍  if in use).